ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -93,11 +94,22 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		// If the JobStatus is not terminal, it is possible that the Ray job is still running. This includes
 		// the case where JobStatus is JobStatusNew.
 		if !rayv1.IsJobTerminal(rayJobInstance.Status.JobStatus) {
+			rayClusterNamespacedName := common.RayJobRayClusterNamespacedName(rayJobInstance)
```

Comment:
We should have a more elegant method to initialize the dashboard client. Calling `GenerateHeadServiceName` every time we want to initialize the client makes the codebase complex and hard to maintain.

Reply:
Agreed, I thought this was kind of ugly too. Maybe we can pass the entire `RayCluster` to `InitClient`?